lib/ash.ex

```diff
@@ -1953,11 +1957,22 @@ defmodule Ash do
   Create a record or raises an error. See `create/2` for more information.
   """
   @doc spark_opts: [{1, @create_opts_schema}]
-  @spec create!(Ash.Changeset.t(), Keyword.t()) ::
+  @spec create!(Ash.Changeset.t() | resource_with_arguments(), Keyword.t()) ::
           Ash.Resource.record()
           | {Ash.Resource.record(), list(Ash.Notifier.Notification.t())}
           | no_return
-  def create!(changeset, opts \\ []) do
+  def create!(changeset_or_resource_with_argument, opts \\ [])
+
+  def create!({resource, arguments}, opts) do
+    action = opts[:action] || Ash.Resource.Info.primary_action!(resource, :create).name
+    arguments = Enum.into(arguments, %{})
+
```

Comment:
Because we construct the changeset here, we also need to pass the relevant opts there. What this would look like is something like:

```elixir
changeset_opts = Keyword.take(opts, Keyword.keys(Ash.Changeset.for_create_opts()))
create_opts = Keyword.take(opts, Keyword.keys(@create_opts_schema))
```

We'll need to do this in each place.

Reply:
There is no `for_update_opts` :thinking:

Reply:
There is no `Ash.Changeset.for_update_opts()` :thinking:

Reply:
We will need to add it then :) It should return the schema used in `Ash.Changeset.for_update`, and should be `@doc false`

Reply:
I allowed myself to:

```elixir
  @for_update_opts @for_create_opts

  @doc false
  def for_update_opts, do: @for_update_opts
```

for now, because I saw no differences :)